tests/test_cmdline_options.py

```diff
@@ -0,0 +1,185 @@
+"""
+Parametrized property-based tests for command line interface.
+
+Contains a strategy, for generating a CLI commands with
+combination of parameters in a randomized order (apart from positional args).
+
+Strategy setup consists of three phases:
+1. Collect available CLI options
+2. Create an atomic strategy for each of those options
+    Each atomic strategy returns list[str] that are appended
+    to the list of arguments to `run_fawltydeps` function.
+3. Create a composite strategy that samples from each atomic
+    strategy and combine results into a new strategy used in the actual tests.
+
+The strategy construction can be viewed as a reference
+to how the CLI options are expected to be used.
+"""
+import string
+from functools import reduce
+from textwrap import dedent
+
+import hypothesis.strategies as st
+from hypothesis import given
+
+from fawltydeps.utils import walk_dir
+from tests.test_cmdline import run_fawltydeps
+
+from .utils import SAMPLE_PROJECTS_DIR
+
+project_with_no_issues = SAMPLE_PROJECTS_DIR / "no_issues"
+CODE_STDIN_MARKER = "-"
+MAX_NUMER_OF_CODE_ARGS = 3
+MAX_NUMER_OF_DEPS_ARGS = 2
+MAX_IGNORE_ARGS = 2
+MAX_VERBOSITY_ARGS = 2
+
+safe_string = st.text(alphabet=string.ascii_letters + string.digits, min_size=1)
+
+# ---------------------------------------------- #
+# CLI options available
+# ---------------------------------------------- #
+actions = [
+    "--check",
+    "--check-undeclared",
+    "--check-unused",
+    "--list-imports",
+    "--list-deps",
+]
+output_formats = ["--summary", "--detailed", "--json"]
+# Options below are  adjusted to `project_with_no_issues`
+# They contain specific paths used in input and a chosen set of dependencies
+# TODO - create more general examples of happy path, including
+#        indicators of missing imports and dependencies
+available_python_input = [
+    str(f)
+    for f in walk_dir(project_with_no_issues)
+    if f.suffix == ".py"
+    # ] + [CODE_STDIN_MARKER]  TODO: uncomment after fixing #233
+]
+# input files currently adjusted to `project_with_no_issues`
+available_deps = [
+    str(project_with_no_issues),
+    str(project_with_no_issues / "requirements.txt"),
+    str(project_with_no_issues / "subdir"),
+]
+# currently adjusted to `project_with_no_issues` (only requirements.txt available)
+deps_parser_choice = [
+    "requirements.txt",
+]
+example_python_stdin = dedent(
+    """\
+    from pprint import pprint
+
+    import pandas as pd
+    import click
+    import tensorflow
+"""
+)
+venvs = [str(project_with_no_issues / ".venv")]
+
+
+# ---------------------------------------------- #
+# Derive strategies from available options
+#
+# Each strategy is of type list[str]
+# ---------------------------------------------- #
+actions_strategy = st.one_of(st.sampled_from(actions), st.none()).map(
+    lambda x: [x] if x is not None else []
+)
```

Comment:
I thought about this and had a bit of conundrum with where to put a `None` or an empty element. 

Using the example of `actions` here:
- One idea is to add it to the list of actions (as you suggest here). Then (as I anticipate, but cannot find a place in hypothesis documentation, where this is stated) the probability of drawing each element from this list is the same. So empty element will occur relatively rare, and in the final, chained generator a probability of choosing a short list of commands will be small.
- Another idea is to put it as a separate strategy (here `st.none()`) that will be drawn half of the time (again, my intuition 🙈 ). In such case, it will be more probable to generate final set of CLI options which is shorter.

Maybe @vreuter could put some more light on it?

Reply:
Have you looked at the approach I suggest in the next comment? It seems that `st.lists(...)` can make all of this _very_ compact. I.e. what you want here is a list of zero or one randomly chosen actions.

Reply:
Yes, and I agree this is more readable and concise  approach. The problem of probability of drawing an empty element remains.

Reply:
After talking with @vreuter we have the same idea of adjusting the probability of choosing the required sample - use `st.none()` as a hack.